### DISCUSSION TITLE
Comment support for YAML files on the Tx

### DISCUSSION THREAD

**Discussion by rdeanchurch** [EdgeTX/edgetx#2140] [Ideas]
Having comments in a modelx.yml file would be handy for explaining some features that are complex... for me.

It appears to me that comments do not hurt a model yml file, but they get stripped out and do not preserve being stored in The MODELS
folder and selected as the current model.  
I went into bootloader mode, plugged the Tx into my PC and modified a model.ylml with a couple of comments then stored it back to the TX SD/MODELS.  I could read it back from the tx and the comments were there.  But after I exited bootloader mode and selected that model and then shut down the tx and went back to bootloader mode and looked at that model.yml again the comments were gone.

Is this of interest to others as a way to make blocks of yml statements use clear?

I'm envisioning documenting what a block of yml statements do.  Then copying them as a method of adding say: 3 timers or dual or triple rates to a new model as a block instead of line by line as in EdgeCompanion or as a bunch of navigation on the tx.

**Comment by pfeerick** [comment]
I don't think it is going to be possible in the short term... the yaml is completely re-written every time it syncs with Companion, and also when changes are made to the model, so any comments that are put in will be stripped out. 

Documenting the file syntax itself is certainly something that will be needed in the future, now that it's actually a format that is human readable. There is also some effort going towards being able to add labels (comments?) to logical switches and special functions that are visible/editable on radio, so that will go somewhat to help with that also. I'm not sure about actual comment blocks though. 

There is also a desire to have a template blocks - i.e. like you said with the timers... which you can then simply reference in multiple models, and then you can make changes in a single place to common elements.

**Comment by rdeanchurch** [comment]
Ahh, I understand the problem now.
Labels on LS and SF will be a help, but not much of one if they are limited to a very few characters.

It does not help much block editing, like copying screenData/"telemetry" pages, telemetry sensors,
rates/expo, a block of ls or sf, or all timers from one model to another.

Thanks for responding
Dean


<http://www.rcgroups.com/forums/member.php?u=139391>
 <http://www.rcgroups.com/forums/member.php?u=139391> <http://www.rcgroups.com/forums/member.php?u=139391>
<http://openrcforums.com/forum/viewtopic.php?f=25&t=185&sid=86567a7056f8c66cb2a565b2f73df302#wrap>
________________________________

**Reply by pfeerick** [reply]
Yes, in this initial implementation it's limited to 10 characters, so limited detail, as he envisaged it might be more a custom name - so I guess the continuation of that would be that it it could be shown instead of `L01` if set... https://github.com/EdgeTX/edgetx/pull/2085

It may just be a matter of tweaking the parsers so that if there are comment blocks within a node to read them in and write them back without any processing... so certainly don't read this as a definite "it won't happen" ... more just "this is why it wipes the stuff out".
